docs/tangle.md

```diff
@@ -0,0 +1,276 @@
+
+Payloads other than transactions are always liked with level of knowledge 3. 
+
+### User-defined payloads
+A node can choose to interpret user-defined payloads by listenting to its specific **payload type** (possibly via third-party code/software). If a node does not know a certain **payload type** it is simply treated as arbitrary data.
+
+### Core payloads
+The core protocol defines a number of payloads that every node needs to interpret and process in order to participate in the network.
+
+- **Transactions:** Value transfers that constitue the ledger state. 
+- **dRNG:** Messages that contain randomness or committee declarations.
+- **FPC:** Opinions on conflicts, mainly issued by high mana nodes.
+
+    
+## Solidification
+A message is said to be **solid** on a node when all its parents are known to the node and also marked as solid.
+
+If a node is missing a referenced message it is stored in the **solidification buffer** and not yet processed. A node can ask its neighbors for the missing message by sending a **solidification request** containing the message hash. This process can be recursively repeated until all of a message's past cone until the genesis (or snapshot) become solid which is known as **solidification**. In that way the Tangle enables even nodes joining the network at a point later in time to retrieve all of a message's history.
+
+### Naive approach
+Approach:
+1. Send solidification request for message immediately to all neighbors
+2. If not received after `solidificationRetryInterval` seconds, send solidification request again. Repeat until received.
+
+This approach simply requests missing messages recursivley until all of them become solid. While easy to implement it has some drawbacks. Each  parent that needs to be requested adds another RTT and message complexity (solidification request * neighbors). 
+
+Optimizations:
+- do not send solidification request immediately: messages can arrive out of order but should generally arrive within a small time window since other nodes only gossip messages on solidification
+- instead of asking for messages one by one a node could ask for messages of a certain timeframe (e.g. its local snapshot time until now)
+- send solidification request only to a subset of neighbors
+
+### Possible attacks
+A malicious node can send unsolidifiable messages to a node. A simple protection is to only repeat solidification requests `solidificationMaxRepeat` times before the message gets deleted from the node's solidification buffer. 
+
+
+## Orphanage
+Messages that are considered to be invalid/not eligible are discarded and are consequently ignored during tip selection. This process of leaving undesired messages (or more in general, whenever a message is not being approved) behind is called **orphaning messages**. It is important that nodes share the same perception on which messages should be orphaned. 
```

Comment:
I'm not sure what does it mean to be discarded. Aren't all messages stored in the local Tangle?